spec/rspec/core/formatters/documentation_formatter_spec.rb

```diff
@@ -81,6 +81,19 @@ def execution_result(values)
 ")
     end
 
+    it "can output indented messages" do
+      root = RSpec.describe("root")
+      root.example("example") {|example| example.reporter.message("message")}
+
+      root.run(reporter)
+
+      expect(formatter_output.string).to eql("
+root
+  message
```

Comment:
Thats odd, I'd actually not expect this output at all...

Reply:
I'd actually expect:
```
root
  example
    message
```
Given your outputting from within the example

Reply:
Yeah, but that's not how it works right now because the example description is only printed after it passes or fails.

Reply:
> I'd actually expect

Isn't because this an explicit usage of "message"?

Thanks for the patch Samuel. We now have probably another patch to submit :)

Reply:
Thinking about this I don't think it makes sense to merge this as is. Indenting incorrectly is just as strange as not indenting at all, possibly worse as its hard to customise. Some more thought is required.

Reply:
1/ It's currently much worse with no indentation.

2/ Indentation "incorrectly" is kind of subjective, since from my POV, it's the correct indentation level (the same as the example). It's just not nested within the example itself.

i.e. unless you can think of something better (completely reworking the documentation formatter), can we merge this and then figure out if there is a better approach?

Reply:
> > 2/ Indentation "incorrectly" is kind of subjective, since from my POV, it's the correct indentation level (the same as the example). 

How can a developer, at a glance tell which example it relates to?

```
root
  example1
  message
  example2
```

> It's just not nested within the example itself.

Which is therefore incorrect given the type of formatter in use.

> i.e. unless you can think of something better (completely reworking the documentation formatter), can we merge this and then figure out if there is a better approach?

I think we need to capture messages into the formatter during example runs, and then output them with the status (which would also need to clear the messages) to get the correct indentation. 

I won't merge a partial solution I'm afraid.